ts/src/gate.ts

```diff
@@ -3626,6 +3625,8 @@ export default class gate extends Exchange {
         } else {
             request['chain'] = currency['id'];
         }
+        const precisionAmount = this.currencyToPrecision (code, amount, network);
```

Comment:
I'm not sure that `currencyToPrecision` will find the `network`, because it has been changed on L3619-3621. Verify why ccxt changes the network here and which format is used in the `currencyToPrecision` (it looks like it extracts networks from currencies, populated in `fetchCurrencies` method)

Reply:
Modifications lines L3619-3621 just test both lowercase and uppercase versions of the networks. If the network key is in uppercase, the network variable will also be in uppercase, if it's not (ie network key is in lowercase or does not exist), the variable will be in lowercase. I tested it